### ISSUE TITLE
Integrations: Integrity constraint violation

### ISSUE THREAD

**Issue by bartdigitalpulse** [solspace/craft3-freeform#721]
**Steps to reproduce**

This error gets regularly reported, about 200 hits in the last 30 days. This is with the Mailchimp integration. The submission is marked as spam.

```
SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails (`db`.`freeform_integrations_queue`, CONSTRAINT `freeform_integrations_queue_id_fk` FOREIGN KEY (`id`) REFERENCES `freeform_mailing_list_fields` (`id`) ON DELETE CASCADE)

The SQL being executed was: 
INSERT INTO `freeform_integrations_queue` (`integrationType`, `status`, `submissionId`, `fieldHash`, `dateCreated`, `dateUpdated`, `uid`) 
VALUES ('mailing_list', 'halted', 51416, 'm52g9Rz1L', '2023-06-28 11:48:27', '2023-06-28 11:48:27', 'ff609026-5d1a-402d-92c0-5e298d41ffac')
```

**Craft & Plugin Info:**

- Craft Version: Craft Pro 3.8.13
- Freeform Version: 3.13.29
- Freeform Edition: Pro
- Fresh Install or Upgrade: No

**Comment by kjmartens** [comment]
Sorry for the trouble @bartdigitalpulse.

This was a fix that we caught and corrected in Freeform 4. However, we'll backport this fix/change over to Freeform 3 as well. I've queued it up, and it should likely be available by early next week.

In the meantime, you could (backup first!) manually remove the `freeform_integrations_queue_id_fk` (Relations) line from the `freeform_integrations_queue` database table. :)

**Comment by kjmartens** [comment]
This is now resolved in Freeform 3.13.30 as well. 🙂
